### PR TITLE
Add cat support for OKR database

### PR DESCRIPTION
**Pull request by MagnusS**
Adds support for an optional OKR database provided as a CSV-file.

Initially also adds support in `cat` for an optional `--okr-db`
parameter. When provided, titles, objectives and projects will be
overwritten in the report with data from the database. If no KR ID is
available the title will be used to look it up in the db.

### CODE REVIEW

**Comment by MagnusS** [comment]
I'll add some tests tomorrow

**Comment by MagnusS** [comment]
This is ready for review. Potential future work here is lint-support and putting the csv path in the conf file

File: src/KR.ml [hunk 1]
```diff
@@ -190,6 +190,28 @@ let make_time_entries t =
   let aux (e, d) = Fmt.strf "@%s (%s)" e (string_of_days d) in
   Item.[ Paragraph (Text (String.concat ", " (List.map aux t))) ]
 
+let update_from_master_db t db =
+  let update (orig_kr : t) (db_kr : Masterdb.elt_t option) =
+    match db_kr with
+    | None -> orig_kr
+    | Some db_kr ->
```
Comment (samoht) [inline src/KR.ml:197]: Maybe add a warning here if the title and objective differ? I think we do this in `merge` and it's quite useful to correct potential mistakes.

Comment (MagnusS) [reply src/KR.ml:197]: Yes, good idea - I added a warning for each modified field

**Pull request merged**